pkg/doc/presexch/definition.go

```diff
@@ -166,3 +168,90 @@ func (pd *PresentationDefinition) ValidateSchema() error {
 
 	return errors.New(strings.Join(errs, ","))
 }
+
+// CreateVP creates verifiable presentation.
+func (pd *PresentationDefinition) CreateVP(credentials ...*verifiable.Credential) (*verifiable.Presentation, error) {
+	if err := pd.ValidateSchema(); err != nil {
+		return nil, err
+	}
+
+	if len(pd.SubmissionRequirements) > 0 {
+		// TODO: handle submission requirements
+		return nil, errors.New("not supported yet")
+	}
+
+	var result []*verifiable.Credential
+
+	for _, descriptor := range pd.InputDescriptors {
+		filtered := filterSchema(descriptor.Schema, credentials)
+		if len(filtered) == 0 {
+			return nil, errors.New("credentials do not satisfy the requirements")
+		}
+
+		result = append(result, filtered...)
+	}
+
+	vp := &verifiable.Presentation{}
+
+	err := vp.SetCredentials(credentialsToInterface(merge(result))...)
+	if err != nil {
+		return nil, err
+	}
+
+	return vp, nil
```

Comment:
This VP is left without context, schema, etc. You'd have to manually set them on line 194.

(we could really use builder APIs for VPs....)

Reply:
VP will be improved. The final result must be a valid VP. We do not have builder APIs :(

Reply:
I don't `presentation_submission` in Verifiable Presentation you generated, is it going to be added in future?

Reply:
@sudeshrshetty yes, it will be added soon